### PR TITLE
Boot PgBouncer after keeper host received

### PR DESCRIPTION
**Pull request by lawrencejones**
This commit changes the way we manage PgBouncer by waiting until a valid
keeper host has been received before starting the PgBouncer process.
This prevents PgBouncer from coming up before we have a valid
destination and should allow readiness health checks to continue
checking for the 6432 tcp port to detect if a pod is route-able.

### CODE REVIEW

**Comment by dyson** [comment]
Merged lawrence-fix-double-component to master and updated the base branch.

File: cmd/stolon-pgbouncer/main.go [hunk 1]
```diff
@@ -490,6 +512,12 @@ func main() {
 								return nil
 							}
 
+							// Let the pgbouncer.child know we've got a value, and that it's ok to boot
+							// PgBouncer. We'll probably fail to reload PgBouncer immediately after the
+							// first time we run this as we'll race PgBouncer to start, however our
+							// subsequent retry will succeed.
+							signalReceivedKeeperHost()
```
Comment (dyson) [inline cmd/stolon-pgbouncer/main.go]: Should we call this after we have generated a new config with the masterAddress?

So at line 530, between `pgBouncer.GenerateConfig` and `pgBouncer.Reload`?

We could then get rid of the initial fake config creation at line 370 as we would never start PgBouncer unless we had a masterAddress and already created a config with it.

Comment (lawrencejones) [reply cmd/stolon-pgbouncer/main.go]: Yeah let's do that

**Comment by lawrencejones** [comment]
Approved on your change @dyson. We good to merge?

**Pull request merged**